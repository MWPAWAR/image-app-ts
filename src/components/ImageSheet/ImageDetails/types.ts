import { ImageTypes } from '../../types'

export interface RowProps {
  label: string
  value: string
}

export interface ImageDetailsProps {
  selectedImage: ImageTypes
}
