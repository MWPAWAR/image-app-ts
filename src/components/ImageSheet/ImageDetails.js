import { getMonthDateAndYear } from '../../utils/date-util';
import imageDetailsStyles from './ImageDetails.module.css';

const Row = ({ label, value }) => (
  <div className={imageDetailsStyles.row}>
    <div className={imageDetailsStyles.label}>{label}</div>
    <div className={imageDetailsStyles.value}>{value}</div>
  </div>
);

const ImageDetails = ({ selectedImage }) => {
  return (
    <div className={imageDetailsStyles.imgDetails}>
      <div className={imageDetailsStyles.informationLabel}>Information</div>
      <Row label="Uploaded by" value={selectedImage.uploadedBy} />
      <Row label="Created" value={getMonthDateAndYear(selectedImage.createdAt)} />
      <Row label="Last modified" value={getMonthDateAndYear(selectedImage.updatedAt)} />
      <Row label="Dimensions" value={`${selectedImage.dimensions.height} X ${selectedImage.dimensions.width}`} />
      <Row label="Resolutions" value={`${selectedImage.resolution.height} X ${selectedImage.resolution.width}`} />
    </div>
  )
};

export default ImageDetails;
