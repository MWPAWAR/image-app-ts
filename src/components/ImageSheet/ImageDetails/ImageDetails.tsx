import { getString } from '../../../i18n'
import { getMonthDateAndYear } from '../../../utils/date-util'
import imageDetailsStyles from './ImageDetails.module.css'
import { RowProps, ImageDetailsProps } from './types'

export const Row = ({ label, value }: RowProps) => (
  <div className={imageDetailsStyles.row}>
    <div className={imageDetailsStyles.label}>{label}</div>
    <div className={imageDetailsStyles.value}>{value}</div>
  </div>
)

const ImageDetails = ({ selectedImage }: ImageDetailsProps) => (
  <div className={imageDetailsStyles.imgDetails}>
    <div className={imageDetailsStyles.informationLabel}>
      {getString('imageSheet.information')}
    </div>
    <Row
      label={getString('imageSheet.uploadedBy')}
      value={selectedImage.uploadedBy}
    />
    <Row
      label={getString('imageSheet.createdAt')}
      value={getMonthDateAndYear(selectedImage.createdAt)}
    />
    <Row
      label={getString('imageSheet.lastModified')}
      value={getMonthDateAndYear(selectedImage.updatedAt)}
    />
    <Row
      label={getString('imageSheet.dimensions')}
      value={`${selectedImage.dimensions.height} X ${selectedImage.dimensions.width}`}
    />
    <Row
      label={getString('imageSheet.resolutions')}
      value={`${selectedImage.resolution.height} X ${selectedImage.resolution.width}`}
    />
  </div>
)

export default ImageDetails
