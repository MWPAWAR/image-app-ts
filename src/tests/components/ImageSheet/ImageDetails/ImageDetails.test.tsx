import renderer from 'react-test-renderer'
import ImageDetails, {
  Row,
} from '../../../../components/ImageSheet/ImageDetails/ImageDetails'
import { imageMock1 } from '../../../mocks/image'

describe('ImageDetails', () => {
  it('renders ImageDeatails correctly', () => {
    const tree = renderer.create(<ImageDetails selectedImage={imageMock1} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders ImageDetails Row correctly', () => {
    const tree = renderer.create(<Row label="ABC" value="XYZ" />)
    expect(tree).toMatchSnapshot()
  })
})
