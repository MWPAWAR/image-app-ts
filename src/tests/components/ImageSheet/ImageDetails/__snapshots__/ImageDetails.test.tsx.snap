// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDetails renders ImageDeatails correctly 1`] = `
<div
  className="imgDetails"
>
  <div
    className="informationLabel"
  >
    Information
  </div>
  <div
    className="row"
  >
    <div
      className="label"
    >
      Uploaded by
    </div>
    <div
      className="value"
    >
      Ms. Jimmie Cole
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="label"
    >
      Created
    </div>
    <div
      className="value"
    >
      July 15, 2017
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="label"
    >
      Last modified
    </div>
    <div
      className="value"
    >
      December 16, 2022
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="label"
    >
      Dimensions
    </div>
    <div
      className="value"
    >
      4800 X 3200
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="label"
    >
      Resolutions
    </div>
    <div
      className="value"
    >
      72 X 72
    </div>
  </div>
</div>
`;

exports[`ImageDetails renders ImageDetails Row correctly 1`] = `
<div
  className="row"
>
  <div
    className="label"
  >
    ABC
  </div>
  <div
    className="value"
  >
    XYZ
  </div>
</div>
`;
